shared/src/protocol/navigation/location.rs

```diff
@@ -0,0 +1,82 @@
+use std::fmt::Display;
+
+use euclid::{approxeq::ApproxEq, Point2D};
+use num_traits::FromPrimitive;
+use serde::{Deserialize, Serialize};
+
+use crate::solar_systems::{self, Solar};
+
+#[derive(Debug, Copy, Hash, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SolarSystemId(pub i64);
+
+impl SolarSystemId {
+    #[must_use]
+    pub fn into_location<T: FromPrimitive>(self) -> Option<T> {
+        T::from_i64(self.0)
+    }
+}
+
+#[cfg(feature = "redis")]
+mod redis {
+    use redis::{FromRedisValue, ToRedisArgs};
+
+    use super::SolarSystemId;
+
+    impl FromRedisValue for SolarSystemId {
+        fn from_redis_value(v: &redis::Value) -> redis::RedisResult<Self> {
+            let value = i64::from_redis_value(v)?;
+            Ok(Self(value))
+        }
+    }
+
+    impl ToRedisArgs for SolarSystemId {
+        fn write_redis_args<W>(&self, out: &mut W)
+        where
+            W: ?Sized + redis::RedisWrite,
+        {
+            self.0.write_redis_args(out)
+        }
+    }
+}
+
+impl Display for SolarSystemId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Pilot {
+    pub system: solar_systems::SolarSystemId,
+    pub location: System,
+}
+
+impl Default for Pilot {
+    fn default() -> Self {
+        Self {
+            system: solar_systems::SolarSystemId::SM0A9F4,
+            location: System::InSpace(Point2D::default()),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum System {
```

Comment:
This isn't a System, it's a Location within a System.

Reply:
I could rename this to `Location`?